File: Data Structures & Algorithms/longest-common-prefix/submission-0.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string check = strs[0];
        string res = "";
        for(int i = 0 ; i < check.size() ; i++)
        {
            for(auto str : strs)
            {
                if(i == str.size() or str[i] != check[i])
                {
                    return res;
                }
                
            }
            res += check[i];
        }
        return res;
    }
};

File: Data Structures & Algorithms/anagram-groups/submission-0.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        groups = defaultdict(list)
        for string in strs:
            count = [0] * 26
            for char in string:
                count[ord(char) - 97] += 1
            groups[tuple(count)].append(string)
        return list(groups.values())
